src/options/append_to_stream.rs

```diff
@@ -0,0 +1,79 @@
+use crate::event_store::client::shared::Empty;
+use crate::event_store::client::streams::append_req::options::ExpectedStreamRevision;
+use crate::{Credentials, EventData, ExpectedVersion};
+use futures::future::Ready;
+use futures::stream::{Iter, Once};
+use futures::Stream;
+
+#[derive(Clone)]
+/// Options of the append to stream command.
+pub struct AppendToStreamOptions {
+    pub(crate) version: ExpectedStreamRevision,
+    pub(crate) credentials: Option<Credentials>,
+}
+
+impl Default for AppendToStreamOptions {
+    fn default() -> Self {
+        Self {
+            version: ExpectedStreamRevision::Any(Empty {}),
+            credentials: None,
+        }
+    }
+}
+
+impl AppendToStreamOptions {
+    /// Performs the command with the given credentials.
+    pub fn authenticated(self, credentials: Credentials) -> Self {
+        Self {
+            credentials: Some(credentials),
+            ..self
+        }
+    }
+
+    /// Asks the server to check that the stream receiving the event is at
+    /// the given expected version. Default: `ExpectedVersion::Any`.
+    pub fn expected_version(self, version: ExpectedVersion) -> Self {
```

Comment:
**SUGGESTION:** Can we use `expected_revision` as it's in other clients and rename type `ExpectedVersion` to `StreamRevision`?

Reply:
Done